ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/remote/RemoteConnectorExecutor.java

```diff
@@ -188,16 +199,65 @@ && getUserRateLimiterMap().get(user.getName()) != null
             if (getMlGuard() != null && !getMlGuard().validate(payload, MLGuard.Type.INPUT)) {
                 throw new IllegalArgumentException("guardrails triggered for user input");
             }
-            invokeRemoteModel(mlInput, parameters, payload, tensorOutputs, countDownLatch, actionListener);
+            if (executionContext.getConnectorRetryOption().isRetryEnabled()) {
+                invokeRemoteModelWithRetry(mlInput, parameters, payload, executionContext, actionListener);
+            } else {
+                invokeRemoteModel(mlInput, parameters, payload, executionContext, actionListener);
+            }
         }
     }
 
+    default void invokeRemoteModelWithRetry(
+        MLInput mlInput,
+        Map<String, String> parameters,
+        String payload,
+        ExecutionContext executionContext,
+        ActionListener<Tuple<Integer, ModelTensors>> actionListener
+    ) {
+        final RetryableAction<Tuple<Integer, ModelTensors>> invokeRemoteModelAction = new RetryableAction<>(
+            getLogger(),
+            getClient().threadPool(),
+            TimeValue.timeValueMillis(executionContext.getConnectorRetryOption().getRetryBackoffMillis()),
+            TimeValue.timeValueSeconds(executionContext.getConnectorRetryOption().getRetryTimeoutSeconds()),
+            actionListener,
+            BackoffPolicy
+                .constantBackoff(
```

Comment:
Is there jitter involved in this constant backoff policy? We should consider making this a configuration as well, different endpoint might need different policy.

Reply:
The jitter is not involved. As mentioned in #2438 , constant backoff is good enough from performance perspective. And since there is still a queue maintained at async client,  the retry request is just put back at the tail of the queue, and not guaranteed to get executed. So I think the constant backoff is good enough.
Can you think of any use cases that we need exponential backoff? I think we can extend current retry options once we find an actual use case or customers call out about it. We already have multiple settings of retry, and I don't want it too complicated. (From my initial thought, the default settings are good enough and it will be rare to tune these settings)

Reply:
It's better to add jitter to avoid all the failure requests are been retried at the same time which can highly possible causing the retry failing. For exponential backoff:
1. Choosing constant or exponential is based on different cases, e.g. if a service doesn't have peak traffic then it's fine to use constant backoff, otherwise an exponential might be more useful since during the peak time, constant can keep failing.
2. I don't think adding an optional configuration to user is adding complication, we can still set default backoff policy to constant.
3. When adding a new feature it's better to think of it a complete one, which means if user wants to use exponential policy, they can simply configure it instead of call out to community. Also, though we're addressing the sagemaker issue initially but we should consider all other cases since this solution is a generic one.

Reply:
OK it makes sense to add a settings for determine here. I'll add a settings to support ExponentialEqualJitterBackoffIterator and ExponentialFullJitterBackoff here

Reply:
For ConstantBackoff, In OpenSearch core we don't have any existing interface to add jitter to it. But I don't think this is a blocking issue. We can make contributions to core and extend the settings here in the future

Reply:
Sure, it's possible that constant backoff was designed to not have jitter, I think the currently policies are enough to cover different cases, if a different policy is needed, we can add it in the future.